app/gui/src/controller/searcher.rs

```diff
@@ -1218,7 +1218,7 @@ impl Searcher {
             let requests = return_types_for_engine.into_iter().map(|return_type| {
                 info!("Requesting suggestions for returnType {return_type:?}.");
                 let file = graph.module.path().file_path();
-                ls.completion(file, &position, &this_type, &return_type, &tags)
+                ls.completion(file, &position, &this_type, &return_type, &tags, &Some(false))
```

Comment:
But now we always filter out static methods, don't we? We should do this only if we have this argument specified.

We should filter out static methods only then `this_type` has a type:
```rust
let this_type = this_type.await;
let is_static = this_type.is_some().then_some(false).
info!("Requesting new suggestion list. Type of `self` is {this_type:?}.");
...
    ls.completion(file, &position, &this_type, &return_type, &tags, &is_static)
```